clippy_lints/src/operators/arithmetic_side_effects.rs

```diff
@@ -11,18 +11,19 @@ use rustc_span::{
     sym,
 };
 
-const HARD_CODED_ALLOWED: &[&str] = &[
-    "&str",
-    "f32",
-    "f64",
-    "std::num::Saturating",
-    "std::num::Wrapping",
-    "std::string::String",
+const HARD_CODED_ALLOWED_BIN: &[&str] = &[
+    "f32,f32",
+    "f64,f64",
+    "std::num::Saturating,std::num::Saturating",
+    "std::num::Wrapping,std::num::Wrapping",
+    "std::string::String,&str",
 ];
+const HARD_CODED_ALLOWED_UNARY: &[&str] = &["f32", "f64", "std::num::Saturating", "std::num::Wrapping"];
 
 #[derive(Debug)]
 pub struct ArithmeticSideEffects {
-    allowed: FxHashSet<String>,
+    allowed_bin: FxHashMap<String, FxHashSet<String>>,
```

Comment:
I might improve this monstrosity in the future but for now, let's just say that it works as intended.